go/host/host.go

```diff
@@ -60,6 +61,9 @@ const (
 
 // Implementation of host.Host.
 type host struct {
+	ctx        context.Context    // context managing the lifespan of this host object
```

Comment:
I read in the docs of the context that you shouldn't store it in a struct. 
not sure why. would be worth investigating

Reply:
Ah good spot. It seems contexts aren't generally used to manage the lifecycle of long-term objects like the host. And tend to be just used for the lifespan of a request or job/process.

I've removed the contexts from the structs and now use a cancelStream mechanism to clean up the streams. It means there's no state on the block provider and it can now be used safely for multiple streams simultaneously I think so that's a benefit :)